### PR TITLE
[eclipse/xtext-core#1042] Make Validation configurable: if entry rule fragement

### PR DESCRIPTION
**Pull request by epoell**
See also [originalIssue](https://github.com/eclipse/xtext-core/issues/1034) [correspondingPR](https://github.com/eclipse/xtext-core/issues/1034)
This will add an menu entry point, whether it should be ignored/warned/marked as error, if the grammar's entry rule is a fragment rule.
The menu entry is placed in the section "Grammar"

Signed-off-by: Eva Poell <epoell@itemis.com>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/validation/messages.properties [hunk 1]
```diff
@@ -8,6 +8,7 @@ XtextValidatorConfigurationBlock_14=The grammar validation settings have changed
 XtextValidatorConfigurationBlock_15=Discouraged rule name:
 XtextValidatorConfigurationBlock_16=Info
 XtextValidatorConfigurationBlock_17=Usage of deprecated rules
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/validation/messages.properties]: Could you please add a colon after 'Usage of deprecated rules', too?

Comment (epoell) [reply org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/validation/messages.properties]: done :)

**Comment by epoell** [comment]
The continuous-integration seems to fail, as the issue_code, that is called in `XtextValidatorConfigurationBlock`: `addComboBox(INVALID_FRAGMENT_AS_FIRST_RULE, Messages.XtextValidatorConfigurationBlock_18, section, defaultIndent);` is uploaded in the Xtext-core project in its own branch.
Can I somehow make the issue_code accessible within this project (at least for the build)?

**Comment by kthoms** [comment]
The upstream repository is https://ci.eclipse.org/xtext/job/xtext-core/view/change-requests/job/PR-1041/lastSuccessfulBuild/artifact/build/p2-repository/
This would need to be changed in the used target file(s) in releng/org.eclipse.xtext.target. And this would need to be reverted before merge.

**Comment by epoell** [comment]
> The upstream repository is https://ci.eclipse.org/xtext/job/xtext-core/view/change-requests/job/PR-1041/lastSuccessfulBuild/artifact/build/p2-repository/
> This would need to be changed in the used target file(s) in releng/org.eclipse.xtext.target. And this would need to be reverted before merge.

Thanks for the hint!
But I'm not sure what the right stream would be. Could you point me in the right direction?

**Comment by kthoms** [comment]
The URL above is the one to insert in the target file where you find `xtext-core`. The default target is oxygen, so change `org.eclipse.xtext.target-oxygen.target`.

**Comment by epoell** [comment]
Thanks! I hope it will build now successfully :crossed_fingers:

**Comment by szarnekow** [comment]
Looks like we do have to rebase the commit on top of master. There are a few unexpected changes.

**Pull request merged**